ntp/syncTime.go

```diff
@@ -73,89 +88,127 @@ type syncTime struct {
 	clockOffset time.Duration
 	syncPeriod  time.Duration
 	ntpOptions  NTPOptions
-	query       func(options NTPOptions) (*ntp.Response, error)
+	query       func(options NTPOptions, hostIndex int) (*ntp.Response, error)
 }
 
-// NewSyncTime creates a syncTime object. The customQueryFunc argument allows
-// the caller to set a different NTP-querying callback, if desired. If set to
-// nil, then the default queryNTP is used.
-func NewSyncTime(ntpConfig config.NTPConfig, syncPeriod time.Duration, customQueryFunc func(options NTPOptions) (*ntp.Response, error)) *syncTime {
-	var queryFunc func(options NTPOptions) (*ntp.Response, error)
-	if customQueryFunc == nil {
+// NewSyncTime creates a syncTime object. The customQueryFunc argument allows the caller to set a different NTP-querying
+// callback, if desired. If set to nil, then the default queryNTP is used
+func NewSyncTime(
+	ntpConfig config.NTPConfig,
+	customQueryFunc func(options NTPOptions, hostIndex int) (*ntp.Response, error),
+) *syncTime {
+	queryFunc := customQueryFunc
+	if queryFunc == nil {
 		queryFunc = queryNTP
-	} else {
-		queryFunc = customQueryFunc
 	}
 
-	hostIndex := checkNTPHost(ntpConfig, queryFunc)
-
 	s := syncTime{
 		clockOffset: 0,
-		syncPeriod:  syncPeriod,
+		syncPeriod:  time.Duration(ntpConfig.SyncPeriodSeconds) * time.Second,
 		query:       queryFunc,
-		ntpOptions:  NewNTPOptions(ntpConfig, hostIndex)}
-
-	return &s
-}
-
-func checkNTPHost(ntpConfig config.NTPConfig, customQueryFunc func(options NTPOptions) (*ntp.Response, error)) int {
-	for hostIndex := range ntpConfig.Hosts {
-		_, err := customQueryFunc(NewNTPOptions(ntpConfig, hostIndex))
-		if err != nil {
-			continue
-		}
-		log.Debug("using NTP server",
-			"host", ntpConfig.Hosts[hostIndex],
-		)
-		return hostIndex
+		ntpOptions:  NewNTPOptions(ntpConfig),
 	}
 
-	return 0
+	return &s
 }
 
-// StartSync method does the time synchronization at every syncPeriod time elapsed. This should be started
-// as a go routine
+// StartSync method does the time synchronization at every syncPeriod time elapsed. This method should be started on go
+// routine
 func (s *syncTime) StartSync() {
 	for {
 		s.sync()
-		time.Sleep(s.syncPeriod)
+		time.Sleep(s.getSleepTime())
 	}
 }
 
-// sync method does the time synchronization and sets the current offset difference between local time
-// and server time with which it has done the synchronization
+func (s *syncTime) getSleepTime() time.Duration {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
you should use "crypto/rand" instead of "math/rand" because will get linter warnings

Reply:
Done